status-im/status-react
# Publish contact updates & contact code periodically

## Contact updates
Currently it's very easy for contact details to get out of sync, the
simplest example is:
A & B are contacts.
A changes name.
B receives the updated name.
B re-install the app.
Until A changes name again, B will not see their name, picture and won't
be able to send push notifications.

This PR changes the behavior to publish account informations to contacts
every 48 hrs, to add some redundancy in this cases.

## Contact code

This PR also publishes contact codes periodically to a public channel, once every 12hrs.
This allow other peers to fetch our information so that they can contact securely using PFS and include all our devices.
This change leaks some metadata, namely if you have been online in the past 12 hours, but it's the same as publishing on swarm/ipfs, we are trading off some darkness for stronger encryption and usability.

### Testing

It's a bit hard to test as the problem only shows after the contact request is not on the mailserver anymore.
Theoretically it goes like this:

1) Create account A
2) Create account B
3) Add B to contacts from A
4) You should now see contact details of A on B
5) Change clock on B to +2 days
6) Unistall/Reinstall  the app on B
7) Login, make sure you are connected, wait until messages are fetched on the mailserver. Check that A is not in your contacts (any message should not have their name). Make sure you can receive messages.
8) Change clock on A +2 days, make sure you are online and connected.
9) Wait a few minutes (+/- 5 mins).

You should see contact information of A on B eventually.

Mind that any time you change the clock, things might get a bit weird (you might go offline/online/offline etc). Generally login/logout fixes it.

status: ready
